addons/website_forum/static/src/js/website_forum.js

```diff
@@ -256,31 +255,24 @@ publicWidget.registry.websiteForum = publicWidget.Widget.extend({
         if (!karma) {
             return;
         }
-
         ev.preventDefault();
-        const forumID = parseInt(document.getElementById('wrapwrap').dataset.forum_id);
-        const notifOptions = {
-            type: "warning",
-            sticky: false,
-        };
         if (session.is_website_user) {
-            notifOptions.title = _t("Access Denied");
-            notifOptions.message =  Markup`
-            <p>Oh no! Please <a href='/web/login'>sign in</a> to vote</p>`;
-        } else {
-            notifOptions.title = _t("Karma Error");
-            // FIXME this translation is bad, the number should be part of the
-            // translation, to fix in the appropriate version
-            notifOptions.message = `${karma} ${_t("karma is required to perform this action. ")}`;
-            if (forumID) {
-                const linkLabel = _t("Read the guidelines to know how to gain karma.");
-                notifOptions.message = Markup`
-                    ${notifOptions.message}<br/>
-                    <a class="alert-link" href="/forum/${encodeURIComponent(forumID)}/faq">${linkLabel}</a>
-                `;
-            }
+            this._displayAccessDeniedNotification(Markup(sprintf(_t('<p>Oh no! Please <a href="%s">sign in</a> to vote</p>'), "/web/login")));
```

Comment:
Can't you extract at least the "p" tag to ease translation ? Like below actually.